File: src/main/java/com/example/venka/javaspringsparkdemo/transformer/MessageTransformer.java
package com.example.venka.javaspringsparkdemo.transformer;

import com.example.venka.javaspringsparkdemo.model.Message;
import org.bson.Document;
import org.bson.types.ObjectId;

public interface MessageTransformer {

    Document toDocument(final Message message);

    Message fromDocument(final Document document);

    Document toDocument(final ObjectId id);
}

File: src/main/kotlin/com/example/venka/kotlinspringsparkdemo/transformer/MessageTransformer.kt
package com.example.venka.kotlinspringsparkdemo.transformer

import com.example.venka.kotlinspringsparkdemo.model.Message
import org.bson.Document
import org.bson.types.ObjectId

interface MessageTransformer {

    fun toDocument(message: Message): Document

    fun fromDocument(document: Document): Message

    fun toDocument(id: ObjectId): Document
}
